pinot-core/src/main/java/org/apache/pinot/core/operator/BaseOperator.java

```diff
@@ -52,4 +54,25 @@ public final T nextBlock() {
 
   // Make it protected because we should always call nextBlock()
   protected abstract T getNextBlock();
+
+  // Enforcing sub-class to implement the getOperatorName(), as they can just return a static final,
+  // as opposed to this super class calling getClass().getSimpleName().
+  public abstract String getOperatorName();
```

Comment:
I thought there was another method we discussed w.r.t whether a node should be skipped or not ?

Reply:
That method was being only used twice (`DocIdSetOperator` and `TransformOperator`), so adding the function at the interface level didn't make sense. The way it is set up now is that if an operator doesn't define a non-null value for getExplainPlanName(), it won't be included in EXPLAIN PLAN output.

```
  /**
   * @return EXPLAIN PLAN name if one is defined; otherwise, null. null EXPLAIN PLAN name means that this operator
   * won't be included in EXPLAIN PLAN output.
   */
  String getExplainPlanName();
```

Reply:
These methods are already included in the interface, no need to redefine them within the base class